include/deal.II/grid/grid_tools.h

```diff
@@ -630,6 +631,15 @@ namespace GridTools
   compute_ghost_cell_halo_layer (const Container &container);
 
 
+  /**
+   * Return the adjacent cells of all the vertices. The vertices are ordered by
+   * their ID.
```

Comment:
Can you rephrase the second sentence by using the term `index` instead of `ID` and then saying that `the vertex index is the number returned by calling <code>cell-@>vertex_index()</code>`? It may also be worthwhile mentioning that not all vertex indices are actually used; the used ones are marked in the array returned by `Triangulation::get_used_vertices()`.